PeriodogramAnalysis/test_power_density_statistics.py

```diff
@@ -0,0 +1,237 @@
+from typing import Generator, Any
+
+import numpy as np
+import pytest
+from collections import defaultdict
+from miv.core.datatype import Signal
+from power_density_statistics import (
+    SpectrumAnalysisBase,
+    SpectrumAnalysisPeriodogram,
+    SpectrumAnalysisWelch,
+)
+
+
+def mock_signal_generator() -> Generator:
+    timestamps = np.linspace(0, 10, 1000, endpoint=False)
+    signal = (
+        np.sin(2 * np.pi * 5 * timestamps)
+        + np.sin(2 * np.pi * 10 * timestamps)
+        + np.sin(2 * np.pi * 15 * timestamps)
+        + np.sin(2 * np.pi * 20 * timestamps)
+    )
+
+    data1 = np.array([signal, signal]).T
+    signal1 = Signal(data=data1, timestamps=timestamps, rate=100)
+    yield signal1
+
+    data2 = np.array([signal, signal]).T
+    signal2 = Signal(data=data2, timestamps=timestamps, rate=100)
+    yield signal2
+
+
+def test_SpectrumAnalysisBase_compute_psd_not_implemented() -> None:
+    """
+    This test is used to test if no subclass is used, will it pop up proper error.
+    """
+    analyzer = SpectrumAnalysisBase()
+    signal = next(mock_signal_generator())
+    psd_dict: dict[int, dict[str, Any]] = {}
+    with pytest.raises(NotImplementedError) as exc_info:
+        analyzer.compute_psd(signal, psd_dict)
+    assert str(exc_info.value) == (
+        "The compute_psd method is not implemented in the base class. "
+        "This base class is not intended for standalone use. Please use a subclass "
+        "such as SpectrumAnalysisWelch, SpectrumAnalysisPeriodogram, or SpectrumAnalysisMultitaper."
+    )
+
+
+def test_SpectrumAnalysisBase_call(mocker) -> None:
+    """
+    This test is first to test if the default value of base class it as expected, second
+    to test if the dict returned by call is correct, if it fails, the call function may be incorrect
+    the functions may not be properly called and the dict's layers might be wrong.
+
+    I mock "compute_psd" and see how many times it is called to see if it is used to handle all chunks.
```

Comment:
Don't mock `compute_psd`, make mock class that inherit `SpectrumAnalysisBase`. Isn't it how user suppose to use?

Reply:
I see, have deleted it, it is not necessary right now.